contents/node/nodejs.md

```diff
@@ -0,0 +1,203 @@
+# Introduction to Node.js
+
+Author: Rachael Sim
+
+# Overview
+
+*Node.js* is a JavaScript runtime built on Chrome’s V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient.
+
+### Event-driven
+
+When an event happens e.g. an I/O event is complete, the event handler/callback function will be enqueued in a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c).
+
+### Non-Blocking
+
+When file and other I/O operations requests are made in Python or Java, they are blocking - subsequent requests (in the same thread) have to be enqueued and can only be processed after the current operation completes. Meanwhile, the program remains idle. In contrast, Node.js has an event loop which delegates the actual tasks to other systems (e.g. file system and databases) and it is almost never blocked after delegation. While an I/O operation is incomplete, the event loop can still process subsequent requests.
```

Comment:
Event driven and non-blocking can be bullet points rather than headings.
>* **Event-driven** : When an event happens e.g. an I/O event is complete, the event handler/callback function will be enqueued in a queue where it will be subsequently scheduled to run by the [event loop](https://medium.com/the-node-js-collection/what-you-should-know-to-really-understand-the-node-js-event-loop-and-its-metrics-c4907b19da4c).